### DISCUSSION TITLE
How the DELETE strategy works?

### DISCUSSION THREAD

**Discussion by javferrod** [mia-platform/community#519] [Q&A] [answered]
### What product are you having troubles with?

Fast Data

### What Console version are you using?

v13

### Description

Hi everyone,

I have a couple of questions about the deletion of projections in Fast Data:

1. What happens to the Single View and Projection record if a delete event of a base projection is captured?
2. What happens to the Single View and Projection record if a delete event of a non-base projection is captured?

I have read [this page](https://docs.mia-platform.eu/docs/fast_data/configuration/single_view_creator/upsert_delete_strategies) of the documentation, which explains the soft and hard delete strategies, but I didn't find anything related to the Single View.

Thanks!

**Comment by ianFar96** [comment, marked as answer]
Hi @javferrod 

The default methods of deletion for Projections and Single Views are Soft Delete for the Projections and Hard Delete for the Single Views. As you mentioned this can be modified by env vars in the services.

Considering the defaults this would be the answer to your questions:

1. What happens to the Single View and Projection record if a delete event of a base projection is captured?
- The projection record will be soft deleted and the single view record will be hard deleted.

2. What happens to the Single View and Projection record if a delete event of a non-base projection is captured?
- the projection record will be soft deleted and the single view will be updated, meaning that the data coming from the projection record will be removed from the single view record and the record will be updated.

I hope this answers you question, let me know if there anything else!

**Reply by javferrod** [reply]
Hi @ianFar96 

Thank you very much for your swift reply. In my case, I'm interested in using the hard delete strategy also for projections. I'm using an older version of Fast Data with the RTU instead of the Projection Storer. I couldn't find how to change the default behavior in the [documentation](https://docs.mia-platform.eu/docs/fast_data/configuration/realtime-updater/configuration-files). Could you please point me in the right direction?

Thanks again.
